File: bin/pad_logo.dart
import 'dart:io';
import 'package:image/image.dart' as img;

void main() {
  try {
    final imageBytes = File('assets/icon/logo.png').readAsBytesSync();
    final image = img.decodeImage(imageBytes);
    if (image == null) {
      print('Could not decode image.');
      return;
    }
    
    final oldW = image.width;
    final oldH = image.height;
    
    final newW = (oldW * 0.65).toInt();
    final newH = (oldH * 0.65).toInt();
    
    final resized = img.copyResize(image, width: newW, height: newH, interpolation: img.Interpolation.cubic);
    
    final padded = img.Image(width: oldW, height: oldH);
    img.fill(padded, color: img.ColorRgba8(0, 0, 0, 0));
    
    final dstX = (oldW - newW) ~/ 2;
    final dstY = (oldH - newH) ~/ 2;
    img.compositeImage(padded, resized, dstX: dstX, dstY: dstY);
    
    File('assets/icon/logo_padded.png').writeAsBytesSync(img.encodePng(padded));
    print('Padded logo created successfully.');
  } catch (e) {
    print('Error: $e');
  }
}

File: pad_logo.py
from PIL import Image

def pad_logo():
    try:
        img = Image.open('assets/icon/logo.png').convert('RGBA')
        old_w, old_h = img.size
        # Shrink to 65% to add safe zone padding
        new_w, new_h = int(old_w * 0.65), int(old_h * 0.65)
        resized_img = img.resize((new_w, new_h), Image.LANCZOS)
        
        # New canvas with transparent background
        new_img = Image.new('RGBA', (old_w, old_h), (0, 0, 0, 0))
        paste_x = (old_w - new_w) // 2
        paste_y = (old_h - new_h) // 2
        new_img.paste(resized_img, (paste_x, paste_y), resized_img)
        
        new_img.save('assets/icon/logo_padded.png')
        print("Padded logo created successfully.")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == '__main__':
    pad_logo()
